avocado/utils/asset.py

```diff
@@ -169,8 +170,11 @@ def fetch(self):
         # Now we have a writable cache_dir. Let's get the asset.
         # Adding the user defined locations to the urls list:
         if self.locations is not None:
-            for item in self.locations:
-                urls.append(item)
+            if isinstance(self.locations, list):
+                for item in self.locations:
+                    urls.append(item)
+            else:
+                urls.append(self.locations)
```

Comment:
What if it's a tuple? :)
The beauty of the current implementation is that it works for any kind of sequence or iterator (lists, tuples, dictionaries, sets, custom iterators, ...).
The only exception is your case: a string. So, you should handle your case without breaking anything else:
```python
    def __init__(...):
        ...
        if isinstance(locations, str):
            self.locations = [locations]
        else:
            self.locations = locations
```
The rest stays the same.

Reply:
@apahim ah  yes right make sense , will do that